src/pages/dataElements/New.spec.tsx

```diff
@@ -0,0 +1,203 @@
+        })
+
+        expect(List).toHaveBeenCalledTimes(0)
+
+        fireEvent.click(cancelButton)
+
+        expect(List).toHaveBeenCalledTimes(1)
+    })
+
+    it('should not submit when required values are missing', async () => {
+        const router = createMemoryRouter([{ path: '/', element: <New /> }])
+        const result = render(
+            <ComponentWithProvider dataForCustomProvider={customData}>
+                <RouterProvider router={router} />
+            </ComponentWithProvider>
+        )
+
+        const submitButton = await result.findByText('Create data element', {
+            selector: 'button',
+        })
+
+        fireEvent.click(submitButton as HTMLButtonElement)
+
+        expect(
+            result.container.querySelectorAll('.error[data-test*="validation"]')
+                .length
+        ).toBe(4)
+
+        expect(
+            result.container.querySelector(
+                '[data-test="dataelementsformfields-name"] .error'
+            )
+        ).toBeTruthy()
+
+        expect(
+            result.container.querySelector(
+                '[data-test="dataelementsformfields-shortname"] .error'
+            )
+        ).toBeTruthy()
+
+        expect(
+            result.container.querySelector(
+                '[data-test="dataelementsformfields-valuetype"] .error'
+            )
+        ).toBeTruthy()
+
+        expect(
+            result.container.querySelector(
+                '[data-test="dataelementsformfields-aggregationtype"] .error'
+            )
+        ).toBeTruthy()
+    })
+
+    it('should submit the data and return to the list view on success', async () => {
+        const List: React.FC = jest.fn(() => <div />)
+        const dataElementCustomData = jest.fn(() => Promise.resolve({}))
+        const router = createMemoryRouter(
+            [
+                { path: '/new', element: <New /> },
+                { path: '/dataElements', element: <List /> },
+            ],
+            {
+                initialIndex: 0,
+                initialEntries: ['/new'],
+            }
+        )
+        const result = render(
+            <>
+                <div id="dhis2-portal-root" />
+                <ComponentWithProvider
+                    dataForCustomProvider={{
+                        ...customData,
+                        dataElements: dataElementCustomData,
+                    }}
+                >
+                    <RouterProvider router={router} />
+                </ComponentWithProvider>
+            </>
+        )
+
+        const submitButton = await result.findByText('Create data element', {
+            selector: 'button',
+        })
+
+        expect(submitButton).toBeTruthy()
+
+        fireEvent.change(
+            result.getByLabelText('Name (required)*') as HTMLElement,
+            { target: { value: 'Data element name' } }
+        )
+
+        fireEvent.change(
+            result.getByLabelText('Short name (required)*') as HTMLElement,
```

Comment:
Unfortunately that wasn't possible. But I still changed it slightly according to [this recommendation](https://testing-library.com/docs/queries/about):

> i. `getByRole`: This can be used to query every element that is exposed in the accessibility tree. With the name option you can filter the returned elements by their accessible name. This should be your top preference for just about everything. There's not much you can't get with this (if you can't, it's possible your UI is inaccessible). Most often, this will be used with the `name` option like so: `getByRole('button', {name: /submit/i})`. Check the list of roles.